### ISSUE TITLE
AudioLevelIndicator stops working on iPad when moving to a different tab

### ISSUE THREAD

**Issue by bojantv** [twilio/twilio-video-app-react#267]
**Describe the bug**
You start the app all is fine the audio level indicator works, you go to another tab, go back and it stops working.

Another issue that I couldn't reproduce on your app because it doesn't have a href to somewhere is:
When we open another tab from the tab playing audio it just shuts off the audio until Safari is restarted. The href is to a link inside our app.
I thought it might be related.

**To Reproduce**
1. Open app, observe audio level indicator working
2. Open a new tab / go to another tab
3. Go back to the app
4. The audio level indicator doesn't indicate anymore

**Expected behavior**
The audio level indicator should keep indicating even after switching tabs.

**Screenshots**

**Environment (please complete the following information):**
 - OS: iOS 13.x.x iPad
 - Browser: Safari
 - App Version: 0.1.0

**Comment by timmydoza** [comment]
Thanks for the detailed issue @bojantv!

I was able to reproduce this issue. Looks like it also happens in desktop Safari.

In a recent pull request, I reworked some things in the [AudioLevelIndicator](https://github.com/twilio/twilio-video-app-react/pull/268/files#diff-34faf33a7e7a2cc0ad4ee95cbfa89cac) component. This seems to fix the issue that you were experiencing. 

Please try it out and let me know how it works for you. Thanks again.

**Comment by bojantv** [comment]
Hi @timmydoza, I appreciate the time to do so. However, after testing what you did, although it works fine in the current app I would like to make some suggestions, and report an "issue" depending on how you look at it.

Like I said in the current app it works fine but if you wanted the audio to be disabled from start, rather then enabled it creates a problem because at the time of cloning the mediaStreamTrack audio needs to be enabled. That causes the analyzer to not initialize correctly and you get no green mic feedback. 

The code I used to test that is:
![image](https://user-images.githubusercontent.com/58706872/88350625-8c66a580-cd4b-11ea-9726-231a0e91ef16.png)
It's just adding newTrack.disable() before it get's set in useLocalTracks() hook. 

The nasty workaround then for that issue is a cloning function that looks like this:
![image](https://user-images.githubusercontent.com/58706872/88350702-ccc62380-cd4b-11ea-978f-855c97628bea.png)
Basically, before cloning it enables the audioTrack, clones the mediaStream, and puts it back to the previous state.
Then it all works.

Another suggestion/issue is one from before (actually it might have been resolved now, did not test it really), is a cleanup of the useEffect() in AudioLevelIndicator in which I would add stoping of all media tracks like this:
![image](https://user-images.githubusercontent.com/58706872/88350948-9dfc7d00-cd4c-11ea-957d-daf11d680de4.png)
Again, in this app it works fine, but in an app with multiple views, many redraws, etc... this can lead to MaxTracks warning which I got a lot before I added similar to the previous code.

I might be completely wrong and missing something, and if I am please let me know.

Thanks.

**Comment by timmydoza** [comment]
Hey @bojantv - thanks for the insights!

I see what you mean about disabling the audio from the start. I also like your addition of `stopAllMediaStreamTracks()` to the cleanup function. 

On my end, I noticed a few instances where the audio level indicator was not animating, even though other participants in the room could hear me just fine. So I have a branch here that seems to fix this issue, and it also allows for the audio track to start out as disabled (which I verified by adding `audioTrack.disable()` like you did).

Let me know what you think: https://github.com/twilio/twilio-video-app-react/compare/audio-indicator-fix

It looks at the `isTrackEnabled` boolean and does not clone tracks and start an analyser when the audio track is disabled.

Also, I haven't seen any MaxTracks warnings. Where did you see those?

**Comment by bojantv** [comment]
Hi @timmydoza, I've checked your branch. It's spot-on, everything seems to be working.

I wasn't able to reproduce Max tracks exceeded warning after your last refactor, I could before. I fixed it by adding the handleStopped() in the cleanup, or I at least like to believe that sorted it.

This is the warning I got before:
![image](https://user-images.githubusercontent.com/58706872/88436758-e2475600-cdfc-11ea-9be6-75b9f1c91efd.png)

Thanks!
